services/user/Evaluations/plugin/wit/world.wit

```diff
@@ -0,0 +1,111 @@
+
+    /// Registers an account for a specific evaluation.
+    /// 
+    /// Requires:
+    /// * Must be called by registrant OR the evaluation owner
+    /// * Must be in registration phase 
+    ///
+    /// Parameters:
+    /// * `evaluation-owner`: The account that owns the evaluation.
+    /// * `evaluation-id`: Unique identifier for the evaluation.
+    /// * `registrant`: The account to be registered.
+    register: func(evaluation-owner: account, evaluation-id: u32, registrant: account) -> result<_, error>;
+
+    /// Unregisters an account from a specific evaluation.
+    /// 
+    /// Requires:
+    /// * Must be in registration phase 
+    ///
+    /// Parameters:
+    /// * `evaluation-owner`: The account that owns the evaluation.
+    /// * `evaluation-id`: Unique identifier for the evaluation.
+    /// * `registrant`: The account to be unregistered.
+    unregister: func(evaluation-owner: account, evaluation-id: u32, registrant: account) -> result<_, error>;
+
+    /// Starts a specific evaluation.
+    ///
+    /// Requires:
+    /// * Callable by anyone
+    /// * Must be in deliberation phase (ideally at the beginning)
+    /// 
+    /// Parameters:
+    /// * `evaluation-id`: Unique identifier for the evaluation.
+    start: func(evaluation-owner: account, evaluation-id: u32) -> result<_, error>;
+
+    /// Closes a specific evaluation
+    ///
+    /// Requires:
+    /// * Must be called by the evaluation owner.
+    /// * Must be in closed phase
+    ///
+    /// Parameters:
+    /// * `evaluation-id`: Unique identifier for the evaluation.
+    close: func(evaluation-owner: account, evaluation-id: u32) -> result<_, error>;
+
+    /// Submits a proposal for a group in an evaluation.
+    ///
+    /// Requires:
+    /// * Callable unlimited times, only during the deliberation phase.
+    ///
+    /// Parameters:
+    /// * `evaluation-owner`: The account that owns the evaluation.
+    /// * `evaluation-id`: Unique identifier for the evaluation.
+    /// * `group-number`: Identifier for the group submitting the proposal.
+    /// * `proposal`: List of `num-options` expressing the users latest vote.
+    propose: func(evaluation-owner: account, evaluation-id: u32, group-number: u32, proposal: list<string>) -> result<_, error>;
+
+    /// Attests participation or completion for a group in an evaluation. Callable only once per group member until result is determined.
+    ///
+    /// Requires:
+    /// * Callable only during the attestation phase.
+    ///
+    /// Parameters:
+    /// * `evaluation-owner`: The account that owns the evaluation.
+    /// * `evaluation-id`: Unique identifier for the evaluation.
+    /// * `group-number`: Identifier for the group attesting.
+    attest: func(evaluation-owner: account, evaluation-id: u32, group-number: u32) -> result<_, error>;
+
+
+    /// Retrieves a proposal for a group in an evaluation.
```

Comment:
Function is confusing, a group doesn't have "a proposal", members have proposals, a group has a result